snapshottest/module.py

```diff
@@ -231,16 +242,19 @@ def assert_value_matches_snapshot(self, test_value, snapshot_value):
     def assert_equals(self, value, snapshot):
         assert value == snapshot
 
-    def assert_match(self, value, name=''):
+    def assert_match(self, value, name="", update=False):
```

Comment:
If I call `assert_match(..., update=False)` I would expect no updates to happen, but if `self.update` is set that would take precedence. This is a bit confusing.

What is the purpose of this parameter? Is it intended only for testing?

If so, I wonder if it would be better for tests to manipulate the `.update` property, rather than add a parameter to this function.

Reply:
I wrote this a while ago, but yeah I think it's only used in tests. I think I did it this way because adding `self.update = True` and `self.update = False` around every test is a lot of boilerplate (it's as many lines of boilerplate as are in most of the tests in the first place).

I definitely agree that it's confusing though. For now I'll rename it to `_force_update_snapshots`, if you think that's ok then you can merge, or if not we can try to think of something else.

Reply:
> I think I did it this way because adding `self.update = True` and `self.update = False` around every test is a lot of boilerplate (it's as many lines of boilerplate as are in most of the tests in the first place).

Does it need to be set back to `False`? I wouldn't think so. Though if it does, what about using a context handler?

Reply:
I had another look: the reason I didn't do it that way was actually because update was a getter function and had some interactions with different testing frameworks that I didn't really understand.

I've pushed a commit with a refactoring that turns it into a plain member variable, and then sets that variable. I think the code is cleaner but I'm less confident that it will work correctly with all supported testing frameworks because I'm not sure I fully understand what the code was doing before (and I don't have anywhere to test django or nose other than the unit tests).

Up to you whether we merge with or without the refactor.

Reply:
I think we probably need to rely on the unit tests, or improve them if they aren't covering the behavior they need to.

What do you imagine might break?

Reply:
I'm worried about "unknown unknowns" because I don't know what I need to worry about with those frameworks (in my experience test frameworks sometimes have magical behaviour, so things can look like they should work but not work).

I think it's probably fine, but maybe we can do an rc or beta build first and get people who are using those frameworks to try it out? Or maybe you're right and we should just trust the unit tests.

Reply:
Sounds good. Let's plan on shipping a bunch of betas until we have tackled all the breaking changes in a release that we've given the community at least a couple weeks to validate.